maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java

```diff
@@ -327,8 +327,11 @@ static String getDefaultGoalPrefix( MavenProject project )
 
         for ( Artifact dependency : project.getArtifacts() )
         {
+            // maven-archiver and maven-jxr are really in wrong groupId
             if ( "org.apache.maven".equals( dependency.getGroupId() )
                 && dependency.getArtifactId().startsWith( "maven-" )
+                && !dependency.getArtifactId().equals( "maven-archiver" )
+                && !dependency.getArtifactId().equals( "maven-jxr" )
```

Comment:
Maybe put excludes names in some list to simplify expression ... 
will be easier add next exclude

Reply:
fixed